sys/vm/vm_mmap.c

```diff
@@ -210,8 +210,8 @@ int
 sys_mmap(struct thread *td, struct mmap_args *uap)
 {
 
-	return (kern_mmap(td, (uintptr_t)uap->addr, uap->len, uap->prot,
-	    uap->flags, uap->fd, uap->pos));
+	return (kern_mmap(td, (__cheri_addr uintptr_t)uap->addr, uap->len,
+	    uap->prot, uap->flags, uap->fd, uap->pos));
```

Comment:
This will need considerably more work as cheriabi_mmap is a rather substantial function.  The mmap family as a whole requires a fair bit of CHERI-specific handling.

Reply:
Yes, this was "just make it compile".  thr_new is another case of something that will need substantial changes.